### PR TITLE
us22+23 Approving apps

### PR DESCRIPTION
**Pull request by d-atkins**
### Add Application Approval
- adds `approved` column to `Application Pets` table (boolean value)
- add `Application Pets Controller`
  - add `update` action
    - sets `approved` to true
    - sets that `ApplicationPet`'s pet's status to "Pending"
- add `Pet` model method: `adopter`

**completes user stories 22 and 23**

### CODE REVIEW

File: app/controllers/application_pets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class ApplicationPetsController < ApplicationController
+  def update
+    application_pet = ApplicationPet.find_by(application_id: params[:application_id], pet_id: params[:pet_id])
```
Comment (d-atkins) [inline app/controllers/application_pets_controller.rb]: finds relevant record in ApplicationPets using the two ids provided by the URI

File: app/controllers/application_pets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    application_pet.update(approved: true)
+    application_pet.pet.update(status: "Pending")
```
Comment (d-atkins) [inline app/controllers/application_pets_controller.rb]: - sets that record's `approved` value to true (this will help the pet know which application it was approved for later on)
- updates that pet's status

File: app/models/pet.rb [hunk 3]
```diff
@@ -19,4 +19,13 @@ def self.pending
   def self.applied_for
     Pet.distinct.joins(:application_pets).order(:id)
   end
+
+  def adopter
+    approved_app = application_pets.find_by(approved: true)
+    if approved_app
+      approved_app.application.name
+    else
+      "nobody"
```
Comment (d-atkins) [inline app/models/pet.rb]: added this edge case in case a pet has its status set to "Pending" without actually being applied for (this is the case for the another model test in `Pet`, shouldn't be possible in the development/production environment)

File: app/views/pets/show.html.erb [hunk 4]
```diff
@@ -8,10 +8,13 @@
...
-    <p>Age: <%= @pet.age %></p>
-    <p>Sex: <%= @pet.sex %></p>
-    <p>Status: <%= @pet.status %></p><br><br>
+    <p id="name">Name: <%= @pet.name %></p>
+    <p id="description">Description: <%= @pet.description %></p>
+    <p id="age">Age: <%= @pet.age %></p>
+    <p id="sex">Sex: <%= @pet.sex %></p>
+    <p id="status">Status: <%= @pet.status %></p><br><br>
+    <% if @pet.status == "Pending" %>
+      <p id="hold">On hold for <%= @pet.adopter %></p>
```
Comment (d-atkins) [inline app/views/pets/show.html.erb]: - added some ids to the pet show, can go back and make tests more robust if we have time (we won't)
- the "On hold for" html only renders if that pet's status is "Pending"

File: config/routes.rb [hunk 5]
```diff
@@ -34,4 +34,6 @@
   get '/applications/new', to: 'applications#new'
   post '/applications', to: 'applications#create'
   get '/applications/:id', to: 'applications#show'
+
+  patch '/applications/:application_id/pets/:pet_id', to: 'application_pets#update'
```
Comment (d-atkins) [inline config/routes.rb]: my reasoning for this routing is that approving an application for a single pet is basically updating that record in `ApplicationPet` - not the application itself. adjusted the URI to include both ids so an ApplicationPet lookup can be made in the controller

File: spec/features/applications/show_spec.rb [hunk 6]
```diff
@@ -58,5 +57,48 @@
       expect(page).to have_link(@pet_2.name)
       expect(page).to have_link(@pet_3.name)
     end
+
+    it "I can approve a pet's application" do
```
Comment (d-atkins) [inline spec/features/applications/show_spec.rb]: made this into a long test - user story 23 (as far as I can tell) was satisfied by default, so I just added onto this test for us22 to cover us23 as well
